File: nomad-samples/common/py/sync-1.1/movies/update_movie.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def update_movie(AUTH_TOKEN, ID, CONTENT_ID, TITLE, SLUG, PLOT, RELEASE_DATE, GENRES, IMAGE_ID, VIDEO_ID):
    # Create header for the request
    if not AUTH_TOKEN:
        raise Exception("Authorization token not found")
  
    API_URL = f"{ADMIN_API_URL}/content/{ID}"
    
    HEADERS = {
      "Content-Type": "application/json",
      "Authorization": "Bearer " + AUTH_TOKEN
    }

    # Build the payload body
    BODY = {
        "contentDefinitionId": MOVIE_CONTENT_DEFINITION_ID,
        "contentId": CONTENT_ID,
        "properties": {},
    }

    if TITLE != "": BODY["properties"]["title"] = TITLE
    if SLUG != "": BODY["properties"]["slugifyField"] = SLUG
    if PLOT != "": BODY["properties"]["plot"] = PLOT
    if RELEASE_DATE != "": BODY["properties"]["releaseDate"] = RELEASE_DATE

    if GENRES != "":
        BODY["properties"]["genres"] = GENRES

    if IMAGE_ID != "":
        BODY["properties"]["image"] = { 
            "id": IMAGE_ID
        }

    if VIDEO_ID != "":
        BODY["properties"]["movieFile"] = {
            "id": VIDEO_ID
        }

    try:
        # Send POST request
        RESPONSE = requests.put(API_URL, headers=HEADERS, data=json.dumps(BODY))

        # Check for success
        if not RESPONSE.ok:
            raise Exception()

        # Get the response
        ID = RESPONSE.json()

        # Return the ID
        return ID
    
    except:
        api_exception_handler(RESPONSE, "Create/Update movie failed")

File: nomad-samples/admin/js/movies/create-movie/movie/update-movie.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function updateMovie(AUTH_TOKEN, ID, TITLE, SLUG, PLOT, RELEASE_DATE, GENRE_MAP, 
                                          IMAGE_ID, VIDEO_ID) {
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);

    // Build the payload body
    const BODY = {
        contentDefinitionId: prjConstants.MOVIE_CONTENT_DEFINITION_ID,
        contentId: ID,
        properties: {}
    };

    if (TITLE !== "") BODY.properties.title = TITLE;
    if (SLUG !== "") BODY.properties.slugifyField = SLUG;
    if (PLOT !== "") BODY.properties.plot = PLOT;
    if (RELEASE_DATE !== "") BODY.properties.releaseDate = RELEASE_DATE;

    if (GENRE_MAP !== "")
    {
        BODY.properties.genres = GENRE_MAP;
    }
    if (IMAGE_ID !== "")
    {
        BODY.properties.image = {};
        BODY.properties.image.id = IMAGE_ID;
        BODY.properties.image.description = `${TITLE} Image`;
    }
    if (VIDEO_ID !== "")
    {
        BODY.properties.movieFile = {};
        BODY.properties.movieFile.id = VIDEO_ID;
        BODY.properties.movieFile.description = `${TITLE} Video`;
    }



    // Send POST request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/content/${ID}`, {
        method: "PUT",
        headers: HEADERS,
        body: JSON.stringify(BODY),
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the response
        const ID = await RESPONSE.text();

        // Return the ID
        return ID;
    }

    apiExceptionHandler(RESPONSE, "Update movie failed")
}
